Extensions/audio_plus.js

```diff
@@ -178,6 +294,34 @@ XKit.extensions.audio_plus = {
 		setTimeout(audio_plus.check_pop_out, 100);
 	},
 
+	waitUntilDockReady: function(docked_video) {
+		var audio_plus = XKit.extensions.audio_plus;
+
+		if (audio_plus.timeout_counter <= 40) { //40 * 50ms = 2s
+			++audio_plus.timeout_counter;
+			console.log(audio_plus.timeout_counter);
+		} else {
+			audio_plus.timeout_counter = 0;
+			clearInterval(audio_plus.waiting_until_dock_ready);
+		}
+
+		if (docked_video.classList.contains("velocity-animating")) {
+			//still animating
+		} else {
+			clearInterval(audio_plus.waiting_until_dock_ready);
+			audio_plus.getDockHeight(docked_video);
+			audio_plus.timeout_counter = 0;
+		}
+	},
+
+	getDockHeight: function(docked_video) {
+		var audio_plus = XKit.extensions.audio_plus;
+
+		var docked_video_height = docked_video.style.height;
+		var controls_style = window.getComputedStyle(audio_plus.pop_out_controls);
+		audio_plus.pop_out_controls.style.transform = "translateY(calc(-" + docked_video_height + " - " + controls_style.bottom + "))";
```

Comment:
so many operands both in this string and constructing it!! much nicer as a string literal:
```js
`translateY(calc(-${docked_video_height} - ${controls_style.bottom}))`
```